gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -589,6 +589,14 @@ private void cancelCurrentRequest() {
     if (resIterator != null) {
       resIterator = null;
     }
+    List<InputStream> unclosedStreams = getObjectMediaResponseMarshaller.popAllStreams();
+    for (InputStream stream : unclosedStreams) {
```

Comment:
Is it possible to create a test for this case ?

Reply:
Unfortunately this isn't feasible to implement until unit tests actually establish connections. Currently we need to rely on our internal tests to verify this.